pages/index.tsx

```diff
@@ -48,35 +47,53 @@ const BugList = ({bugs}: {bugs: any[]}) => {
 
 
 const Home: NextPage = (props) => {
+  const [winReady, setwinReady] = useState(false);
+  useEffect(() => {
+      setwinReady(true);
+  }, []);
 
-  const [state, setState] = useState({bugs: props.data});
+  const [state, setState] = useState(props.data);
 
-  
-  
   function onDragEnd(result: any) {
-    console.log({result, state})
+    console.log({state})
     if (!result.destination) {
       return;
     }
 
     if (result.destination.index === result.source.index) {
       return;
     }
-
+    
     if (result.destination.droppableId === result.source.droppableId) {
+      let destinationColumn = state[result.destination.droppableId]
       const bugs = reorder(
-        state.bugs,
+        destinationColumn.list,
         result.source.index,
         result.destination.index
       );
-
-      setState({ bugs });
+      let newCol = {
+        id: destinationColumn.id,
+        list: bugs
+      }
+
+      console.log("my object", { 
+        ...state,
+        [destinationColumn.id]: {
+          ...newCol
+        }
+      },
+      "newCol: ", {newCol}
+      )
+      
+      setState({ 
+        ...state,
+        [destinationColumn.id]: {
+          ...newCol
+        }
+    });
```

Comment:
re-creates a column and sets state with previous state spread and that column set with the new column replacing the previous column.

effectively this reorders the bug list